src/python/turicreate/toolkits/_mps_utils.py

```diff
@@ -419,7 +423,7 @@ def __init__(self, network_id):
         self._cur_learning_rate = None
 
     def __del__(self):
-        self._LIB.TCMPSDeleteGraphModule(self.handle)
+        self._LIB.TCMPSDeleteGraphModule(self.handle, self.network_id)
 
     def init(self, n, c_in, h_in, w_in, c_out, h_out, w_out, config=None, weights=None):
```

Comment:
Not exactly sure why we'd move it into `__init__`. Don't we need to free the pointer once we're done using it? If we delete it in `__init__` we can't use the pointer to do training for inference, since that handle would now not be referencing anything useful.

Reply:
Sorry, I meant to ask if we could merge the `__init__` and `init` functions

Reply:
Ohhhhhh got it!